src/store/home/actions.js

```diff
@@ -0,0 +1,100 @@
+import axios from "axios";
+import { apiUrl } from "../../config/constants";
+import { selectUser } from "../../store/user/selectors";
+import {
+  // appLoading,
+  appDoneLoading,
+  // showMessageWithTimeout,
+  // setMessage,
+} from "../appState/actions";
+export const FETCH_ALL_POSTS_SUCCESS = "FETCH_POSTS_SUCCESS";
+export const ADD_A_COMMENT = "ADD_A_COMMENT";
+export const DELETE_A_POST = "DELETE_A_POST";
+const fetchPostsSuccess = (posts) => ({
+  type: FETCH_ALL_POSTS_SUCCESS,
+  payload: posts,
+});
+
+export const fetchAllPosts = () => {
+  return async (dispatch, getState) => {
+    try {
+      const response = await axios.get(`${apiUrl}/posts/all`);
+      dispatch(fetchPostsSuccess(response.data));
+    } catch (error) {
+      console.log("error", error);
+    }
+  };
+};
+// Fetch one post
+export const FETCH_A_POST_SUCCESS = "FETCH_A_POST_SUCCESS";
```

Comment:
keep an order in your action files, generally it goes like:
1. Action Types (like this one)
2. Action Creators (like fetchAPostSuccess)
3. Thunks